### ISSUE TITLE
APF_levy test fails

### ISSUE THREAD

**Issue by kbwestfall** [pypeit/PypeIt#307]
I think this is a problem with pypit, not the development suite, so I'm posting it here.

There is a problem when testing the APF_levy script.  I ran the following script:

```
#!/usr/bin/env python3

import sys
import os
import time
import numpy
import subprocess

#-----------------------------------------------------------------------------

#-----------------------------------------------------------------------------

if __name__ == '__main__':
    t = time.clock()

    pwd = os.getcwd()
    print(pwd)

    instr = 'APF_levy'

    raw_data = 'RAW_DATA'
    outputdir = 'REDUX_OUT'

    wdir = os.path.join(pwd, outputdir, instr)
    if not os.path.isdir(wdir):
        os.makedirs(wdir)

    redfile = os.path.join(pwd, 'pypit_files', '{0}.pypit'.format(instr.lower()))
    if not os.path.exists(redfile):
        raise FileNotFoundError('No file {0}'.redfile)

    # Edit the reduce file
    with open(redfile, 'r') as infile:
        lines = infile.readlines()

    for kk, iline in enumerate(lines):
        if 'data read' in iline:
            j = kk + 1
            while True:
                if lines[j][0] != '#':
                    break
                j += 1
            dpth = lines[j]
            i0 = dpth.rfind('/')
            newdpth = ' {0}'.format(os.path.join(pwd, raw_data, instr))+dpth[i0:]
            lines[j] = newdpth
        elif 'reduce flatfield useframe' in iline:
            cpth = lines[kk]
            i0 = cpth.rfind('/')
            newcpth = ' {0}'.format(os.path.join(pwd, 'CALIBS', cpth[i0:]))
            lines[kk] = 'reduce flatfield useframe '+newcpth

    # Write new reduce file to output directory
    pyp_file = '{0}.pypit'.format(instr.lower())
    outfile = os.path.join(wdir, pyp_file)
    print(outfile)
    with open(outfile, 'w') as ofile:
        for iline in lines:
            ofile.writelines(iline)

    logfile = os.path.join(wdir, '{0}.log'.format(instr.lower()))
    with open(logfile, 'w') as f:
        print(['run_pypit', pyp_file, '--devtest', '-v 2'])
        print(wdir)
        retval = subprocess.call(['run_pypit', pyp_file, '--devtest', '-v 2'], stderr=f, cwd=wdir)

        if retval == 0:
            print("\x1B[" + "1;32m" + "PASSED" + "\x1B[" + "0m", file=sys.stderr)
            npass += 1
        else:
            print("\x1B[" + "1;31m" + "FAILED" + "\x1B[" + "0m", file=sys.stderr)
        subprocess.call(['tail', '-2', logfile])
        print("\n", file=sys.stderr)

    print('Elapsed time: {0} seconds'.format(time.clock() - t))

```

It failed, so I reran it using:

`run_pypit apf_levy.pypit --develop`

and got the following traceback:

```
Traceback (most recent call last):
  File "/Library/Frameworks/Python.framework/Versions/3.6/bin/run_pypit", line 6, in <module>
    exec(compile(open(__file__).read(), __file__, 'exec'))
  File "/Users/westfall/Work/packages/PYPIT/bin/run_pypit", line 17, in <module>
    rtval = run_pypit.main(args)
  File "/Users/westfall/Work/packages/PYPIT/pypit/scripts/run_pypit.py", line 84, in main
    use_masters=args.use_masters, devtest=args.devtest, logname=logname, debug=debug)
  File "/Users/westfall/Work/packages/PYPIT/pypit/pypit.py", line 246, in PYPIT
    status = armed.ARMED(fitsdict)
  File "/Users/westfall/Work/packages/PYPIT/pypit/armed.py", line 133, in ARMED
    pcadesc="PCA trace of the slit edges")
  File "/Users/westfall/Work/packages/PYPIT/pypit/artrace.py", line 1946, in trace_slits
    skipx0=False, function=settings.argflag['trace']['slits']['function'])
  File "/Users/westfall/Work/packages/PYPIT/pypit/arpca.py", line 73, in basis
    sub = (yfit - high_order_fit) * outmask
ValueError: operands could not be broadcast together with shapes (4608,76) (4608,75)
```

**Comment by rcooke-ast** [comment]
The code is picking up (or missing) 1 slit. Definitely a problem with PYPIT and not the development suite. I’ll take a look.

**Comment by profxj** [comment]
I have identified a work-around in the
fuss_with_apf branch

**Comment by profxj** [comment]
And I can confirm it gets the code far enough in the Development suite
to run the arcytrace.expand_slits method (and its underlying Cython).

@kbwestfall -- please go forth to deprecate that Cython code.

**Comment by kbwestfall** [comment]
A straight conversion of expand_slits to python is *slow* (2.5 sec vs. 0.04 sec).  I'm playing around with speeding it up, but I'm unclear about some expected behavior.  @rcooke-ast, what happens when `msedge[x,y] ` is never greater than `edgv` in the `y` loops?  I seems like it could happen if the difference between, say, `msedge[x,ordcen[x,o]]` is different enough from `msedge[x,ordcen[x,o]] - 1` when searching left.  And I think I'm seeing this behavior in the APF_levy test, but I need to confirm...  If this is happening, should we be setting `mwid = 1`?

**Comment by rcooke-ast** [comment]
Good catch! It should be masked in this case, and set to -1.  Thanks.

**Comment by profxj** [comment]
Given that master is failing anyways for APF, I think it is fine
to make the PR from this branch.

**Comment by rcooke-ast** [comment]
Yes, indeed - I'll also take a look at some point to clean up the APF reduction a little.

**Comment by rcooke-ast** [comment]
Appears to be addressed by PR #327 - we also need to come back to this when the echelle pipeline is in full force.
